Libraries/Utilities/AlertIOS.js

```diff
@@ -27,61 +26,160 @@ export type AlertButtonStyle = $Enum<{
+ * iOS-specific functionality.
  *
- * Use this API for iOS-specific features, such as prompting the user to enter
- * some information. In other cases, especially to show static alerts, use
- * the cross-platform `Alert` API.
+ * `AlertIOS.prompt` allows you to prompt the user for input inside of an
+ * alert popup.
  *
- * ```
- * AlertIOS.alert(
- *   'Enter password',
- *   null,
- *   [
- *     {text: 'Submit', onPress: (text) => console.log('Password: ' + text)},
- *   ],
- *   'secure-text'
- * )
- * ```
  */
 class AlertIOS {
+  /**
+   * Creates a popup to alert the user. See
+   * [Alert](/react-native/docs/alert.html).
+   *
+   * Arguments:
+   *
+   *  - **title**: string -- The dialog's title.
+   *  - **message**: string -- An optional message that appears above the text input.
+   *  - **callbackOrButtons** -- This optional argument should be either a
+   *    single-argument function or an array of buttons. If passed a function,
+   *    it will be called when the user taps 'OK'.
+   *
+   *    If passed an array of button configurations, each button should include
+   *    a `text` key, as well as optional `onPress` and `style` keys.
+   *    `style` should be one of 'default', 'cancel' or 'destructive'.
+   *  - **type** -- *deprecated, do not use*
+   *
+   * Example:
+   *
+   * ```
+   * AlertIOS.alert(
+   *  'Sync Complete',
+   *  'All your data are belong to us.'
+   * );
+   * ```
+   */
   static alert(
     title: ?string,
     message?: ?string,
-    buttons?: Array<{
-      text?: string;
-      onPress?: ?Function;
-      style?: AlertButtonStyle;
-    }>,
-    type?: ?AlertType
+    callbackOrButtons?: ?(() => void) | ButtonsArray,
+    type?: AlertType,
   ): void {
+    if (typeof type !== 'undefined') {
+      console.warn('AlertIOS.alert() with a 4th "type" parameter is deprecated and will be removed. Use AlertIOS.prompt() instead.');
+      this.prompt(title, message, callbackOrButtons, type);
+      return;
+    }
+    this.prompt(title, message, callbackOrButtons, 'default');
+  }
+
+  /**
+   * Prompt the user to enter some text.
+   *
+   * Arguments:
+   *
+   *  - **title**: string -- The dialog's title.
+   *  - **message**: string -- An optional message that appears above the text input.
+   *  - **callbackOrButtons** -- This optional argument should be either a
+   *    single-argument function or an array of buttons. If passed a function,
+   *    it will be called with the prompt's value when the user taps 'OK'.
+   *
+   *    If passed an array of button configurations, each button should include
+   *    a `text` key, as well as optional `onPress` and `style` keys (see example).
+   *    `style` should be one of 'default', 'cancel' or 'destructive'.
+   *  - **type**: string -- This configures the text input. One of 'plain-text',
+   *    'secure-text' or 'login-password'.
+   *  - **defaultValue**: string -- the default value for the text field.
```

Comment:
nit: I like the emphasis for the names but not sure it's a convention in the docs. Check other files. Would rather keep it consistent with docs for other files and then migrate all at once.